lib/utils/remote-build.ts

```diff
@@ -105,7 +107,11 @@ export async function startRemoteBuild(build: RemoteBuild): Promise<void> {
 		stream.on('data', getBuilderMessageHandler(build));
 		stream.on('end', resolve);
 		stream.on('error', reject);
-	}).return();
+	}).then(() => {
+		if (build.hadError) {
+			throw new Error('Remote build failed');
```

Comment:
This seems good, but I'd like to also ensure we don't send all these errors to Sentry as well. There's an `expectedError` helper for exactly that, which prints the normal exception handling user error text, then kills the process without reporting anything.

We should probably do that in the action though, not here... Sound sensible? If so, I think you want to:

* Throw a custom error class here, like `BuildFailedError`
* Catch that error in the action, and consider it expected instead

Reply:
Ah, I forgot about the `exitWithExpectedError` function! Yeah that sounds good, will do.